server/admin_management/api/app/repositories/application_admin_repository.py

```diff
@@ -65,3 +66,26 @@ def delete_application_admin(self, application_admin_id: int):
         )
         self.db.delete(record)
         self.db.flush()
+
+    def get_user_app_admin_grants(self, user_id: int) -> List[FamApplication]:
+        """
+        Find out from `app_fam.fam_application_admin` the applications
+            being granted for the user; including "FAM" application.
+
+        Filter on: Only 'IDIR' type user can be an Application Admin.
+
+        :param user_id: primary id that is associated with the user.
+        :return: List of "applications" granted for the user or None.
+        """
+        return (
+            self.db.query(FamApplication)
+                .select_from(FamApplicationAdmin)
+                .join(FamApplicationAdmin.application)
+                .join(FamApplicationAdmin.user)
+                .filter(
+                    FamApplicationAdmin.user_id == user_id,
+                    FamUser.user_type_code == UserType.IDIR)
```

Comment:
As above, shouldn't be filtering here on IDIR - such entries shouldn't exist in the first place.

Reply:
Yes, agree, db should not this record happen; but I still put the filter to be safe. As even right now it could happen since our create function did not have the validation in place yet.
I can remove it if prefer.